File: backend/ai-job-matcher/src/main/java/com/aijobmatcher/service/AuthService.java
package com.aijobmatcher.service;

import com.aijobmatcher.dto.LoginRequest;
import com.aijobmatcher.dto.LoginResponse;
import com.aijobmatcher.dto.RegisterRequest;

public interface AuthService {

	void register(RegisterRequest request);
	
	LoginResponse login(LoginRequest request);
}

File: frontend/ai-job-matcher-ui/src/app/core/services/auth.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, tap } from 'rxjs';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  token: string;
}

@Injectable({
  providedIn: 'root',
})
export class AuthService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = 'http://localhost:8080/api/auth';
  private readonly tokenKey = 'ai-job-matcher-token';

  login(request: LoginRequest): Observable<LoginResponse> {
    return this.http.post<LoginResponse>(`${this.apiUrl}/login`, request).pipe(
      tap((response) => localStorage.setItem(this.tokenKey, response.token)),
    );
  }

  getToken(): string | null {
    return localStorage.getItem(this.tokenKey);
  }

  logout(): void {
    localStorage.removeItem(this.tokenKey);
  }
}
